File: modules/_foundation/src/main/java/com/projecthub/base/project/domain/event/ProjectStatusChangedEvent.java
package com.projecthub.base.project.domain.event;

import lombok.Value;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Event published when a project's workflow status changes.
 */
@Value
public class ProjectStatusChangedEvent {
    UUID projectId;
    String fromStatus;
    String toStatus;
    LocalDateTime timestamp;
}

File: src/main/kotlin/com/projecthub/domain/project/event/ProjectStatusChangedEvent.kt
package com.projecthub.domain.project.event

import java.time.LocalDateTime

/**
 * Event that is published when a project's status changes.
 * This event can be consumed by other components to react to project status changes.
 */
data class ProjectStatusChangedEvent(
    val projectId: String,
    val previousStatus: String,
    val newStatus: String,
    val timestamp: LocalDateTime
)